include/kdbush.hpp

```diff
@@ -1,64 +1,89 @@
+// https://github.com/mourner/kdbush.hpp
+
 #pragma once
 
 #include <algorithm>
 #include <cmath>
 #include <vector>
 
+template<class TFloat = double, class TIndex = std::size_t>
```

Comment:
I don't have strong feelings about whether it's TCoord or TFloat. Personally I'd expect a TCoord to be some kind of 2D or 3D tuple of X, Y (and Z), but it's not that important since it's pretty obvious what's going on. You're right in that TFloat doesn't describe integer values very well. Which has me agreeing that TCoord is better.

As for comparing it with boost:rtree or alternatives. No, I haven't done that. I know your work through your excellent Javascript packages, and when we recently dropped a javascript (typescript actually) project in favor of a C++ rewrite, I was pleasantly surprised you had ported one of your libraries over to C++.

Even better, you had done it in precisely the single-header minimalist kind of way that I love in both Javascript as well as C++. So I didn't really look anywhere else, because your stuff is small, independent, fast, elegant, and a joy to use.

By the way, I recommend you add a link to your library to this list:

   https://github.com/nothings/stb/blob/master/docs/other_libs.md

It's sort of the go-to source for tiny libraries such as yours.

Thanks again,

Jaap Suter - http://jaapsuter.com